storage/pin/pin.go

```diff
@@ -0,0 +1,400 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pin
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"sync"
+
+	"github.com/ethersphere/swarm/api"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/state"
+	"github.com/ethersphere/swarm/storage"
+	"github.com/ethersphere/swarm/storage/localstore"
+)
+
+const (
+	Version            = "1.0"
+	WorkerChanSize     = 8             // Max no of goroutines when walking the file tree
+	SwarmPinHeaderName = "x-swarm-pin" // Presence of this in header indicates pinning required
```

Comment:
See it form the HTTP header perspective. If you suddenly see a HeaderName variable in the HTTP packet.. does it not look out of place..?

Reply:
It might look, right, so maybe to name it HTTPHeaderName. It is in the pin package and it is a part of the swarm project, so this information is redundant in variable name. I have no strong opinions on that, I am fine if you want to leave the variable as it is.

Reply:
`HeaderName` indeed @janos is right. 
same with `tag.HeaderName` for TODO on tags :)

Reply:
Done

Reply:
why is this not in the `HTTP` server together with the Tag header name?

Reply:
So that all things pinning can belong to /storage/pin package. It looks clean if we use "pin.HeaderName" in the places we use them. Also check what @zelig suggested for tags TODO above.